lib/ansible/galaxy/token.py

```diff
@@ -87,9 +199,31 @@ def get(self):
 
         data = json.loads(to_text(resp.read(), errors='surrogate_or_strict'))
 
-        # - extract 'access_token'
+        if self.jwks is None and self._jwks_url is not None:
+            jwks_resp = open_url(
+                to_native(self._jwks_url),
+                data=payload,
+                validate_certs=self.validate_certs,
+                method='GET',
+                http_agent=user_agent()
+            )
+            self.jwks = json.loads(to_text(jwks_resp.read(), errors='surrogate_or_strict')).get('keys', [])
+
+        # - extract token details
         self._token = data.get('access_token')
 
+        if self.jwks is not None and HAS_CRYPTO:
```

Comment:
Why not fail on no crypto? I'd expect it to be mandatory for this to work...

Reply:
This was for backwards compatibility since the token is used without validation on devel. Failing on no crypto is maybe fine since it's required for ansible-core, but it wasn't necessary before for ansible-galaxy (similarly, I started using `packaging` recently in `ansible-galaxy` but it caused some inconvenience even though it was already a core dep https://github.com/ansible/ansible/issues/78275). So if I'm doing that, I'd want to make a note of it in the changelog. As it is now, only the fix is dependent on crypto, and without it the behavior will stay the same (i.e. the token will be used until it expires).